### ISSUE TITLE
'options' => CarbonInterface::ROUND causes browser/request hang.

### ISSUE THREAD

**Issue by marbuser** [briannesbitt/Carbon#2136]
Hello,

I encountered an issue with the following code:
```php
$date->diffForHumans(['parts' => 1, 'options' => CarbonInterface::ROUND]);
```

Carbon version: 2.36

PHP version: 7.4.7

I expected to get:

```
A rounded month. E.g 1 month 28 days = 2 months.
```

But I actually get:

```
The browser hangs and times out on any page that uses this function call. If I remove `'options' => CarbonInterface::ROUND` it works perfectly fine again. So it seems to be related to that option. On the "try-carbon" heroku app it appears to hang as well.
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

The following perfectly works on try-carbon:

```php
echo Carbon::now()->subMonths(1)->subDays(28)->diffForHumans(['parts' => 1, 'options' => CarbonInterface::ROUND]);
```

It outputs "2 months" instantly.

Please provide a reproductible example.

Thanks.

**Comment by marbuser** [comment]
@kylekatarnls 

Sure no problem. Here;
```php
$date = '2020-07-30 13:51:15';

echo Carbon::parse($date)->diffForHumans(['options' => CarbonInterface::ROUND]);
```

If you try this on https://try-carbon.herokuapp.com/ it hangs and will never re-evaluate the expression. If you remove `['options' => CarbonInterface::ROUND]` from it, it re-evaluates perfectly.

Also tried this in other environments such as Tinkerwell (PHP playground app), and just other projects I'm doing currently. Same result happens in all of them.

The code you provided does indeed work for me as well, so perhaps this is an issue with the `diffForHumans` method and the `parse` method being used together?

Because if I do exactly the same code posted above but with the `now()` method instead, it works perfectly.
```php
Carbon::now()->diffForHumans(['options' => CarbonInterface::ROUND]);
```

**Comment by kylekatarnls** [comment]
Thank you, it will be fixed for the release 2.37.0.
